update_employees.py

```diff
@@ -287,9 +286,12 @@ def build_payload(
     :param user_role_field: field in knack app to specify Staff, Supervisor etc
     :param email_field: field in knack for email
     :param name_field: field in knack for user's first and last name
+    :param result: dict to contain log of changes
     :return:
     """
     payload = []
+    result["updates"] = []
+    result["additions"] = []
```

Comment:
since you're just passing in an empty dict to `build_payload()`, the dependencies on this dict might be more clear if you create the results object entirely here, and then later return it with the payload:

```python
def build_payload():
    result = { "updates":  [], "additions": [], "errors": [] }
    # ....
    return payload, result

# later on...
payload, result = build_payload(...)
```

Reply:
@chiaberry edited after i re-read the code 🙈